svm/tests/integration_test.rs

```diff
@@ -1085,7 +1086,692 @@ fn simple_nonce(enable_fee_only_transactions: bool, fee_paying_nonce: bool) -> V
+    {
+        let mut test_entry = SvmTestEntry::default();
+
+        let source_keypair = Keypair::new();
+        let source = source_keypair.pubkey();
+        let destination1 = Pubkey::new_unique();
+        let destination2 = Pubkey::new_unique();
+
+        let mut source_data = AccountSharedData::default();
+        let destination1_data = AccountSharedData::default();
+        let destination2_data = AccountSharedData::default();
+
+        source_data.set_lamports(LAMPORTS_PER_SOL * 10);
+        test_entry.add_initial_account(source, &source_data);
+
+        for (destination, mut destination_data) in [
+            (destination1, destination1_data),
+            (destination2, destination2_data),
+        ] {
+            test_entry.push_transaction(system_transaction::transfer(
+                &source_keypair,
+                &destination,
+                transfer_amount,
+                Hash::default(),
+            ));
+
+            destination_data
+                .checked_add_lamports(transfer_amount)
+                .unwrap();
+            test_entry.create_expected_account(destination, &destination_data);
+
+            test_entry
+                .decrease_expected_lamports(&source, transfer_amount + LAMPORTS_PER_SIGNATURE);
+        }
+
+        test_entries.push(test_entry);
+    }
+
+    // batch 1:
+    // * successful transfer, source left with rent-exempt minimum
+    // * non-processable transfer due to underfunded fee-payer
+    {
+        let mut test_entry = SvmTestEntry::default();
+
+        let source_keypair = Keypair::new();
+        let source = source_keypair.pubkey();
+        let destination = Pubkey::new_unique();
+
+        let mut source_data = AccountSharedData::default();
+        let mut destination_data = AccountSharedData::default();
+
+        source_data.set_lamports(transfer_amount + LAMPORTS_PER_SIGNATURE + wallet_rent);
+        test_entry.add_initial_account(source, &source_data);
+
+        test_entry.push_transaction(system_transaction::transfer(
+            &source_keypair,
+            &destination,
+            transfer_amount,
+            Hash::default(),
+        ));
+
+        destination_data
+            .checked_add_lamports(transfer_amount)
+            .unwrap();
+        test_entry.create_expected_account(destination, &destination_data);
+
+        test_entry.decrease_expected_lamports(&source, transfer_amount + LAMPORTS_PER_SIGNATURE);
+
+        test_entry.push_transaction_with_status(
+            system_transaction::transfer(
+                &source_keypair,
+                &destination,
+                transfer_amount,
+                Hash::default(),
+            ),
+            ExecutionStatus::Discarded,
+        );
+
+        test_entries.push(test_entry);
+    }
+
+    // batch 2:
+    // * successful transfer to a previously unfunded account
+    // * successful transfer using the new account as a fee-payer in the same batch
```

Comment:
great to test here, commenting a quick thought.
Likely we will rarely see this case in practice. Before scheduling scheduler will do a quick fee-payer check and will drop there before sending to workers.